files/default/neutron-ha-tool.py

```diff
@@ -654,10 +654,14 @@ def migrate_router(qclient, router_id, agent, target,
     qclient.remove_router_from_l3_agent(agent['id'], router_id)
     LOG.debug("Removed router from agent=%s" % agent['id'])
 
-    # ensure it is removed or log an error
+    # ensure it is removed
     if router_id in list_routers_on_l3_agent(qclient, agent['id']):
-        raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent['id']))
+        qclient.remove_router_from_l3_agent(agent['id'], router_id)
+        LOG.debug("Removed (again) router from agent=%s" % agent['id'])
```

Comment:
> Actually the resource agent is retrying that, and when I did some manual test this worked. So that means that if I am not wrong, the first delete raised and exception somewhere.

During the upgrade we're not calling neutron-ha-tool from the resource agent. And the behaviour of ```--retry``` actually interferes with other requirements of the upgrade as it would also retry in cases where we need to raise an error during the ugprade (see https://bugzilla.suse.com/show_bug.cgi?id=1020914 that's why we also introduced: https://github.com/SUSE-Cloud/cookbook-openstack-network/pull/20 ).